examples/video_multipart_create.py

```diff
@@ -2,53 +2,117 @@
 # -*- coding: utf-8 -*-
 
 import logging
-
-import jwplatform
+import os
 import requests
 
+from jwplatform import Client
+from jwplatform.errors import JWPlatformError
+
+
 logging.basicConfig(level=logging.INFO)
 
+JW_API_KEY = os.environ.get('JW_API_KEY')
+JW_API_SECRET = os.environ.get('JW_API_SECRET')
 
-def create_video(api_key, api_secret, local_video_path, api_format='json', **kwargs):
+BYTES_TO_BUFFER = 10000000
+
+
+def run_upload(video_file_path):
     """
-    Function which creates new video object via multipart upload method.
-
-    :param api_key: <string> JWPlatform api-key
-    :param api_secret: <string> JWPlatform shared-secret
-    :param local_video_path: <string> Path to media on local machine.
-    :param api_format: <string> REQUIRED Acceptable values include 'py','xml','json',and 'php'
-    :param kwargs: Arguments conforming to standards found @ https://developer.jwplayer.com/jw-platform/reference/v1/methods/videos/create.html
-    :return:
+    Configures all of the needed upload_parameters and sets up all information pertinent
+    to the video to be uploaded.
+
+    :param video_file_path: <str> the absolute path to the video file
     """
-    # Setup API client
-    jwplatform_client = jwplatform.Client(api_key, api_secret)
 
-    # Make /videos/create API call
-    logging.info("Registering new Video-Object")
     try:
-        response = jwplatform_client.videos.create(upload_method='multipart', **kwargs)
-    except jwplatform.errors.JWPlatformError as e:
-        logging.error("Encountered an error creating a video\n{}".format(e))
-    logging.info(response)
+        upload_parameters = {
+            'file_path': video_file_path,
+            'file_size': os.stat(video_file_path).st_size,
+            'file_name': os.path.basename(video_file_path)
+        }
+
+    except OSError:
+        logging.error('Invalid file path for video file')
+        raise
+
+    try:
+        # Setup API client
+        jwplatform_client = Client(JW_API_KEY, JW_API_SECRET)
+
+        # Make /videos/create API call with multipart parameter specified
+        jwplatform_video_create_response = jwplatform_client.videos.create(
+            upload_method='multipart',
+            title=upload_parameters['file_name']
+        )
+
+    except JWPlatformError:
+        logging.error('An error occurred during the uploader setup. Check that your API keys are properly '
+                      'set up in your environment, and ensure that the video file path exists.')
+        raise
 
     # Construct base url for upload
-    upload_url = '{}://{}{}'.format(
-        response['link']['protocol'],
-        response['link']['address'],
-        response['link']['path']
+    upload_parameters['upload_url'] = '{}://{}{}'.format(
```

Comment:
Might want to use a more explicit format string: `'{protocol}://{address}{path}'.format(**jwplatform_video_create_response['link'])`
Not sure whether this would improve or decrease readability...